### PR TITLE
Fix mpi4py required

### LINKED ISSUE #2211: Setting ADIOS2_USE_Python=ON doesn't error when mpi4py is not found
**Describe the bug**
If `ADIOS2_USE_Python` is `ON`, and `ADIOS2_USE_MPI` is `ON`, but `mpi4py` is not present, configuration does not error like it should.

**To Reproduce**
1. Make sure Python development files are installed but `mpi4py` is not
2. Configure with `ADIOS2_USE_Python=ON` and `ADIOS2_USE_MPI=ON`
3. Observe that no error occurs

**Expected behavior**
An error should occur, since the required `mpi4py` was not found

**Additional context**
This appears to be due to how `find_package(PythonModule)` works. It handles variables for `PythonModule_<module>` rather than `PythonModule`. So when `REQUIRED` is passed, `PythonModule_FIND_REQUIRED` is set to true, but `find_package_handle_standard_args()` reads `PythonModule_mpi4py_FIND_REQUIRED`, which is not set, and so the error is not triggered.

I believe that the solution is to either change how `find_package(PythonModule)` works, or write a dedicated find module for `mpi4py`.

### PR DESCRIPTION
**Pull request by chuckatkins**
Fixes #2211

### CODE REVIEW

File: cmake/FindPythonModule.cmake [hunk 1]
```diff
@@ -121,12 +121,19 @@ set( PythonModule_${module_NAME}_PATH
   endif()
 
   include(FindPackageHandleStandardArgs)
-  set(FPHSA_NAME_MISMATCHED TRUE)
+  set(CMAKE_FIND_PACKAGE_NAME PythonModule_${module_NAME})
+  foreach(VAR IN ITEMS REQUIRED QUIETLY VERSION COMPONENTS)
```
Comment (KyleFromKitware) [inline cmake/FindPythonModule.cmake:125]: There are other variables as well. Whenever I need to do something like this, I refer to [this list](https://cmake.org/cmake/help/latest/command/find_package.html#id6).

Comment (chuckatkins) [reply cmake/FindPythonModule.cmake:125]: I went through the list and the individual components are handled differently so those are intentionally skipped.  The remaining are specific version components which should be okay just handling the whole version variable, right?

Comment (KyleFromKitware) [reply cmake/FindPythonModule.cmake:125]: If it works as is then I guess that's OK.

**Comment by liangwang0734** [comment]
Sorry for referring back to this: how can we build the MPI adios2 with Python but without mpi4py?

**Comment by dmitry-ganyushin** [comment]
Yes, but you need to build ADIOS2 without MPI:
cmake options:  -DADIOS2_USE_MPI=OFF    -DADIOS2_USE_Python=ON
-DPYTHON_EXECUTABLE=/usr/bin/python3

**Pull request merged**